### PR TITLE
Filter course metrics by course IDs

### PR DESCRIPTION
**Pull request by marcospri**
For:

- https://github.com/hypothesis/lms/issues/6418

### CODE REVIEW

File: lms/views/dashboard/api/course.py [hunk 1]
```diff
@@ -70,18 +73,21 @@ def courses(self) -> APICourses:
...
+        courses_query = self.course_service.get_courses(
+            organization=org,
+            instructor_h_userid=self.request.user.h_userid
+            if self.request.user
+            else None,
+            h_userids=filter_by_h_userids,
+            assignment_ids=filter_by_h_assignment_ids,
+        )
+        if filter_by_course_ids:
+            courses_query = courses_query.where(Course.id.in_(filter_by_course_ids))
```
Comment (robertknight) [inline lms/views/dashboard/api/course.py:88]: What is the policy for deciding whether to do the filtering inside `get_courses` or outside?

**Review by robertknight — COMMENTED** [review]
This looks quite straightforward. I had a query about filtering inside vs outside of `get_courses`.

File: lms/views/dashboard/api/course.py (hunk 1 above)
Comment (marcospri) [reply lms/views/dashboard/api/course.py:88]: I don't think there's clear policy.

The first and often best candidate is the service, we can abstract any complexity about models/queries there.

In this case I reckon the code reads better: we get a complex query for all the relevant rows from the service and filter it down immediately by a list of IDs from the parameter the view parameter. Those IDs are the PK of the table we are more interested, there's no joins or any extra complexity.

I reckon there's a risk of a future performance "improvement" if looking at the query in isolation in the service. When filtering a set of rows by the PK of in the same table we could say, just get the rows with this IDs and avoid doing all the other fluff, but that will be of course wrong.


That being said I don't have a strong opinion either way.

**Comment by marcospri** [comment]
> marcospri closed this 6 minutes ago

Not sure what happen there :thinking: , I must have done some strange rebasing and GH thought there were no commits? 

Anyway, re-opened it.

**Review by robertknight — APPROVED** [review]

**Pull request merged**